### ISSUE TITLE
Slow training

### ISSUE THREAD

**Issue by Bk073** [herobd/Visual-Template-Free-Form-Parsing#10]
Training the model is extremely slow. Is there anyway to make the training iteration fast? How long did it take for you to train detector network for 150,000 iterations and pairing network for 125,000  iterations?

**Comment by herobd** [comment]
Yes, training is slow. It took me a couple days for the detector. The pairing network takes even longer, on the order of a couple weeks.
I'll also point you to my follow-on work, as it has a stronger model. However, I train it even longer. It takes about a month. I'd highly recommend fine-tuning a pre-trained model to your specific task.

http://arxiv.org/abs/2105.08194
https://github.com/herobd/FUDGE

Part of the reason for the slowness is getting the GT aligned to the predictions. In the code for FUDGE this has been sped up. FUDGE introduces more slowdowns with graph editing. However, if you use the non-graph (`binary`) version of FUDGE (which is very close to this repos model) it should be a little faster.
